frontend/packages/ceph-storage-plugin/src/components/dashboards/persistent-internal/raw-capacity-card/raw-capacity-card.tsx

```diff
@@ -10,11 +10,34 @@ import DashboardCardBody from '@console/shared/src/components/dashboard/dashboar
 import DashboardCardHeader from '@console/shared/src/components/dashboard/dashboard-card/DashboardCardHeader';
 import { FieldLevelHelp, humanizeBinaryBytes } from '@console/internal/components/utils';
 import { getInstantVectorStats } from '@console/internal/components/graphs/utils';
+import {
+  RedExclamationCircleIcon,
+  YellowExclamationTriangleIcon,
+} from '@console/shared/src/components/status/icons';
 import { CAPACITY_INFO_QUERIES } from '../../../../queries';
+import {
+  DANGER,
+  WARNING,
+  DANGER_THRESHOLD,
+  WARNING_THRESHOLD,
+} from '../../common/capacity-breakdown/consts';
 import './raw-capacity-card.scss';
 
 const queries = (() => Object.values(CAPACITY_INFO_QUERIES))();
 const colorScale = ['#0166cc', '#d6d6d6'];
+let capacityStatus = '';
+
+const setCapacityStatus = (capacityRatio) => {
+  if (capacityRatio > DANGER_THRESHOLD) {
+    capacityStatus = DANGER;
+    colorScale[0] = '#C9190B';
+  } else if (capacityRatio > WARNING_THRESHOLD && capacityRatio <= DANGER_THRESHOLD) {
+    capacityStatus = WARNING;
+    colorScale[0] = '#F0AB00';
+  } else {
+    capacityStatus = '';
+  }
+};
```

Comment:
move them inside of the component. Make color scale a React state. Set capacity state should run inside a React useEffect. Should run every time `capacityRatio` changes.